### ISSUE TITLE
Unable to import Euterpea

### ISSUE THREAD

**Issue by sbandx** [Euterpea/Euterpea2#46]
MacMini:bin user$ ghci
GHCi, version 8.8.4: https://www.haskell.org/ghc/  :? for help
Loaded package environment from /Users/serge/.ghc/x86_64-darwin-8.8.4/environments/default
Prelude> import Euterpea

<no location info>: error:
    Could not find module ‘Euterpea’

**Comment by ZachFontenot** [comment]
There are some answers in regard to this in some of the other posts, but you'll need to run Euterpea by setting up your .cabal file to import the library as a dependency and use the old-style or v1 cabal commands to install and run the repl, so *cabal v1-install* and *cabal v1-repl*. Hope that helps

**Comment by donya** [comment]
Euterpea may not work with GHCi, version 8.8.4; the last stable tested verison was 8.6. See recommended GHC versions by OS here: http://www.euterpea.com

**Comment by nxg** [comment]
Unfortunately, I think that versions 8.6.5 and 8.4.3 (as recommended on [the installation page](https://www.euterpea.com/download-and-installation/) cannot really be said to work for a current macOS version – I'm running 11.6 (‘Big Sur'; not the _most_ recent version, but close enough).  I've downloaded them from what I believe to be the recommended download page (eg for [8.4.3](https://www.haskell.org/ghc/download_ghc_8_4_3.html)).

With both of them, `./configure --prefix=foo` works fine, but `make install` fails when it tries to install eg `ghc-cabal`, with the ‘unsigned code‘ error (`“ghc-cabal” cannot be opened because the developer cannot be verified.`), which appears for the various `.dylib` files the installation wants to install, too.  While it's clearly possible to use the ‘Security & Privacy’ Preference pane to override this, it has to be done one by one for (clearly) a long list of dylibs, or else I'd have to disable the relevant protections globally with `spctl`.  I don't really want to do either of those, in the first case because it's a long list (!), and in the second case because `spctl` links into a subsystem I don't know a lot about, so I'd have to do a fair amount of reading before I felt comfortable monkeying with it.  In each case, it seems clear that there's significant version-skew between current Euterpea and current macOS, which makes me... nervous.

Incidentally, trying to install a more recent version (8.10.7) does fail as expected with

    % cabal install Euterpea
    Resolving dependencies...
    cabal: Could not resolve dependencies:
    [__0] trying: Euterpea-2.0.7 (user goal)
    [__1] next goal: Euterpea:setup.Cabal (dependency of Euterpea)
    [__1] rejecting: Euterpea:setup.Cabal-3.2.1.0/installed-3.2.1.0 (conflict:
    Euterpea => Euterpea:setup.Cabal>=1.7 && <1.25)
    ....

(I'm noting this here as a comment to an existing but old ‘can't install’ issue – would it make more sense for me to create a new issue?)

**Comment by donya** [comment]
On ghc 8.10.7 (via ghcup) it installs for me on Windows 10 using a local copy of the Euterpea2 GitHub repo with:

cabal v1-install --allow-newer
(run from within the folder containing Euterpea.cabal)

I do not have access to a Mac to do testing with any version of MacOS at this time. If using ghc 8.10.7 please try with the v1-install command and --allow-newer flag.

**Comment by nxg** [comment]
Yes – thanks! – the `v1-install` command does work for me, iff `--allow-newer` is present.  I found notes from the last time I tried this, which also arose from advice here, completely deleted the install I had (which seemed to have got into an unexpected state), reinstalled ghcup from scratch, and was able to get to a sound!

Regarding this particular issue, I think it might be worth retiring or qualifying the advice on [the installation page](https://www.euterpea.com/download-and-installation/), that versions 8.4 or 8.6 will work on current macOS.

Also, as a point of documentation, can I suggest it may be worth repeating the advice to use `--allow-newer` on that installation page.  I appreciate that it's mentioned on the ‘getting started’ page, but by the time I'd arrived at the page of installation details, the getting-started page was disappearing into the confused fog of my rear-view mirror.
